ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -624,17 +606,14 @@ public boolean hasFailed() {
       return true;
     }
     if (routerOperation == RouterOperation.PutOperation && routerConfig.routerPutUseDynamicSuccessTarget
-        && diskReplicasPresent) {
+        && replicasPresent) {
       return totalReplicaCount - failedCount < Math.max(totalReplicaCount - 1,
           routerConfig.routerPutSuccessTarget + disabledCount);
     } else {
-      // if there is no possible way to use the remaining replicas to meet either the disk or cloud success target,
+      // if there is no possible way to use the remaining replicas to meet the success target,
       // deem the operation a failure.
-      if (!diskReplicasPresent || diskReplicaInPoolOrFlightCount + diskReplicaSuccessCount < diskReplicaSuccessTarget) {
-        if (!cloudReplicasPresent
-            || cloudReplicaInPoolOrFlightCount + cloudReplicaSuccessCount < cloudReplicaSuccessTarget) {
-          return true;
-        }
+      if (!replicasPresent || replicaInPoolOrFlightCount + replicaSuccessCount < replicaSuccessTarget) {
```

Comment:
same here.

Reply:
Done in 
212d358
Can you take another look?